### PR TITLE
update addon develop guide

### PR DESCRIPTION
**Pull request by haoqing0110**
Ref: https://github.com/open-cluster-management-io/ocm/issues/355

### CODE REVIEW

File: content/en/developer-guides/addon.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ This page is a developer guide about how to build an OCM add-on using addon-fram
 
 <!-- spellchecker-enable -->
 
+## Supported version
+
+The OCM v0.14.0 requires addon-framework v0.8.1 and above versions.
```
Comment (qiujian16) [inline content/en/developer-guides/addon.md:16]: we have not released yet. This annoucement should be noted only we are about to release

File: content/en/developer-guides/addon.md [hunk 2]
```diff
@@ -394,6 +404,34 @@ agentAddon, err := addonfactory.NewAgentAddonFactory(addonName, FS, "manifests")
                     BuildTemplateAgentAddon()
 ```
 
+Addtionally, need to grant a `patch` permission on `ClusterManagementAddon` to your addon manager.
```
Comment (qiujian16) [inline content/en/developer-guides/addon.md:407]: This is the requirement for 0.8.1, right?

File: content/en/developer-guides/addon.md (hunk 1 above)
Comment (haoqing0110) [reply content/en/developer-guides/addon.md]: modified the doc to only focus on the released version. will create and hold another PR for v0.14.0.

File: content/en/developer-guides/addon.md (hunk 2 above)
Comment (haoqing0110) [reply content/en/developer-guides/addon.md]: Yes, and 0.9.0 as well. Have updated the content.

File: content/en/developer-guides/addon.md [hunk 3]
```diff
@@ -394,6 +399,34 @@ agentAddon, err := addonfactory.NewAgentAddonFactory(addonName, FS, "manifests")
                     BuildTemplateAgentAddon()
 ```
 
+Addtionally, if you are using addon-framework v0.8.1 and v0.9.0, need to grant a `patch` permission on `ClusterManagementAddon` to your addon manager.
```
Comment (qiujian16) [inline content/en/developer-guides/addon.md:402]:
```suggestion
Addtionally, if you are using addon-framework v0.8.1 or higher, need to grant a `patch` permission on `ClusterManagementAddon` to your addon manager.
```

**Pull request merged**